gammapy/utils/tests/test_parallel.py

```diff
@@ -25,3 +30,80 @@ def func(arg):
         parallel.run_multiprocessing(
             func, [True, True], method="wrong_name", pool_kwargs=dict(processes=2)
         )
+
+
+def square(x):
+    return x**2
+
+
+class MyTask:
+    def __init__(self):
+        self.sum_squared = 0
+
+    def __call__(self, x):
+        return x**2
+
+    def callback(self, result):
+        self.sum_squared += result
+
+
+def test_run_multiprocessing_simple_starmap():
+    N = 10
+    inputs = [(_,) for _ in range(N + 1)]
+
+    result = parallel.run_multiprocessing(
+        func=square,
+        inputs=inputs,
+        method="starmap",
+        pool_kwargs=dict(processes=2),
+    )
+    assert sum(result) == N * (N + 1) * (2 * N + 1) / 6
```

Comment:
A little fun fact here: the analytical solution was actually proposed correctly by Copilot...